runtime/browser/runtime_geolocation_permission_context.cc

```diff
@@ -82,6 +101,47 @@ RuntimeGeolocationPermissionContext::RequestGeolocationPermission(
           web_contents,
           requesting_frame,
           result_callback));
+#else
+  DCHECK(content::BrowserThread::CurrentlyOn(content::BrowserThread::UI));
+  XWalkPermissionDialogManager* permission_dialog_manager =
+      XWalkPermissionDialogManager::FromWebContents(web_contents);
+  if (!permission_dialog_manager) {
+    XWalkPermissionDialogManager::CreateForWebContents(web_contents);
+    permission_dialog_manager =
+        XWalkPermissionDialogManager::FromWebContents(web_contents);
+  }
+
+  PrefService* pref_service =
+      user_prefs::UserPrefs::Get(XWalkBrowserContext::GetDefault());
+  base::string16 text = l10n_util::GetStringFUTF16(
+      IDS_GEOLOCATION_DIALOG_QUESTION,
+      // FIXME : We should get the application name from the manifest.
+      url_formatter::FormatUrl(
+      requesting_frame, pref_service->GetString(kIntlAcceptLanguage),
+      url_formatter::kFormatUrlOmitUsernamePassword |
+      url_formatter::kFormatUrlOmitTrailingSlashOnBareHostname,
+      net::UnescapeRule::SPACES, nullptr, nullptr, nullptr));
+
+  permission_dialog_manager->RequestPermission(
+      CONTENT_SETTINGS_TYPE_GEOLOCATION,
+      requesting_frame, "", text,
+      base::Bind(
+      &RuntimeGeolocationPermissionContext::OnPermissionRequestFinished,
+      base::Unretained(this), result_callback));
```

Comment:
Not sure but it doesn't matter because both are linked to web_contents, if the later gets destroyed, permission_dialog_manager is going to shutdown and the callback will never be called if that's your concern.